common/beerocks/bcl/source/beerocks_socket_thread.cpp

```diff
@@ -256,6 +256,15 @@ bool socket_thread::work()
 {
     before_select();
 
+    auto awake_time_msec = std::chrono::duration_cast<std::chrono::milliseconds>(
+                               std::chrono::steady_clock::now() - m_select_wake_up_time)
+                               .count();
+
+    if (m_select_timeout_msec > 0 && awake_time_msec > m_select_timeout_msec) {
+        THREAD_LOG(WARNING) << "Thread awake time is exceptionally long: " << int(awake_time_msec)
```

Comment:
You should probably print the actual timeout value, otherwise it's a bit hard to understand why did you decide that the awake time is exceptionally long.

Another interesting feature we can add is an average processing time of the thread.
For example, store processing times for 10sec (or 30sec), calculate the average value and print it.

Reply:
I'm don't think that it is interesting what is the select timeout and how we decide what is "exceptionally long".
IMO the printing of how much time we are awake is the only thing matters.
Regarding the average value calculation. I don't think that printing the average will help us...

Reply:
The whole purpose of log messages is to inform the reader about progress and/or issues.
In this case, printing the actual timeout will help the user to quickly understand the problem.